x/dsmr/storage.go

```diff
@@ -326,6 +356,15 @@ func (s *ChunkStorage[T]) GetChunkBytes(expiry int64, chunkID ids.ID) ([]byte, e
 	return chunkBytes, nil
 }
 
+func (s *ChunkStorage[T]) CheckRateLimit(chunk Chunk[T]) error {
+	weightLimit := s.ruleFactory.GetRules(chunk.Expiry).GetMaxAccumulatedProducerChunkWeight()
+
+	if uint64(len(chunk.bytes))+s.pendingChunksSizes[chunk.Producer] > weightLimit {
+		return ErrChunkRateLimitSurpassed
+	}
+	return nil
+}
```

Comment:
Can we add this to complete the existing TODO here: https://github.com/ava-labs/hypersdk/blob/cdd2245a718539cccf4eb25a23058e086e243639/x/dsmr/storage.go#L84 rather than add this into storage?